### ISSUE TITLE
Find a better mechanism to define the dirty-state of the model

### ISSUE THREAD

**Issue by planger** [eclipsesource/graphical-lsp#62]
In the fix for #36 we mark a model as dirty as soon as an `executeOperation_*` is sent to the server. However, this is not a very reliable mechanism. Without having thought this through entirely, imho, the server needs to decide whether a model is dirty or not; eventually, the server decides which action is actually performed or not, isn't it?

**Comment by CamilleLetavernier** [comment]
> Without having thought this through entirely, imho, the server needs to decide whether a model is dirty or not; eventually, the server decides which action is actually performed or not, isn't it?

I agree; especially when the "hints" aren't enough, the client doesn't have enough information to know if an edition operation will be accepted or rejected, or even if it's valid but doesn't actually modify the model (e.g. navigate a link...)

**Comment by planger** [comment]
This is related to #231. Once we have a command stack on the server, it'll be easy for the server to define the dirty-state of the diagram.
